dragonfly/item/glass_bottle.go

```diff
@@ -0,0 +1,30 @@
+package item
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/internal/item_internal"
+	"github.com/df-mc/dragonfly/dragonfly/item/potion"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/go-gl/mathgl/mgl64"
+)
+
+// GlassBottle is an item that can hold various liquids.
+type GlassBottle struct{}
+
+// UseOnBlock ...
+func (g GlassBottle) UseOnBlock(pos world.BlockPos, _ world.Face, _ mgl64.Vec3, w *world.World, _ User, ctx *UseContext) bool {
+	if liquid, ok := w.Liquid(pos); ok {
+		if item_internal.IsWater(liquid) && liquid.LiquidDepth() == 8 {
+			if liquid == w.Block(pos) {
```

Comment:
Couldn't you just do `item_internal.IsWater(w.Block(pos))` and make that function accept a `world.Block`, then try to assert that to `block.Water` in that function?